sdcm/provision/aws/utils.py

```diff
@@ -0,0 +1,364 @@
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation; either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright (c) 2021 ScyllaDB
+
+import abc
+import contextlib
+import datetime
+import time
+from textwrap import dedent
+from typing import Any, Callable, List, Dict, Optional
+
+import boto3
+from botocore.exceptions import ClientError
+from mypy_boto3_ec2 import EC2ServiceResource, EC2Client
+from mypy_boto3_ec2.service_resource import Instance
+from mypy_boto3_ec2.type_defs import InstanceTypeDef, SpotFleetLaunchSpecificationTypeDef, \
+    RequestSpotLaunchSpecificationTypeDef, SpotFleetRequestConfigDataTypeDef
+
+from sdcm.provision.common.provisioner import TagsType
+
+
+SPOT_CNT_LIMIT = 20
```

Comment:
Why is it 20? Where does it come from?

Reply:
We have it in cluster_aws.py from the begging of the time, i.e. 2017

Reply:
Ok :)
I don’t remember what it means, can you please add there a comment?